core/src/main/java/org/elasticsearch/snapshots/SnapshotInfo.java

```diff
@@ -35,52 +39,87 @@
 import java.util.List;
 
 /**
- * Information about snapshot
+ * Information about a snapshot
  */
-public class SnapshotInfo implements ToXContent, Streamable {
+public final class SnapshotInfo implements Comparable<SnapshotInfo>, ToXContent, FromXContentBuilder<SnapshotInfo>, Writeable {
```

Comment:
The issue is, the `ChecksumBlobStoreFormat` depends on getting an object that implements `FromXContentBuilder`